test/widgets/dialog_checks.dart

```diff
@@ -46,12 +51,47 @@ Widget checkErrorDialog(WidgetTester tester, {
   }
 }
 
-/// Checks that there is no dialog.
-/// Fails if one is found.
+
+/// In a widget test, check that there aren't any alert dialogs.
+///
+/// See also:
+///  - [checkErrorDialog]
 void checkNoDialog(WidgetTester tester) {
-  check(find.byType(Dialog)).findsNothing();
-  check(find.bySubtype<AlertDialog>()).findsNothing();
-  check(find.byType(CupertinoAlertDialog)).findsNothing();
+  final List<Widget> alertDialogs = [
+    ...find.bySubtype<AlertDialog>().evaluate().map((e) => e.widget),
+    ...find.byType(CupertinoAlertDialog).evaluate().map((e) => e.widget),
+  ];
+
+  if (alertDialogs.isNotEmpty) {
+    String message = 'Found dialog(s) when none were expected:\n';
+    for (final alertDialog in alertDialogs) {
+      final (title, content) = switch (alertDialog) {
+        AlertDialog()          => (alertDialog.title, alertDialog.content),
+        CupertinoAlertDialog() => (alertDialog.title, alertDialog.content),
+        _ => throw UnimplementedError(),
+      };
+
+      message += 'Dialog:\n'
+                 '  title: ${title is Text ? title.data : title.toString()}\n';
+
+      if (content != null) {
+        final contentTexts = tester.widgetList<Text>(find.descendant(
+          matchRoot: true,
+          of: find.byWidget(content),
+          matching: find.byType(Text)));
+        message += '  content: ';
+        if (contentTexts.isNotEmpty) {
+          message += contentTexts.map((t) => t.data).join('\n    ');
```

Comment:
Best to use a StringBuffer rather than `+=` on a String.

(This isn't a performance-critical spot here; but it's not significantly harder and sets a good example.)